kubernetes/resource_kubernetes_env.go

```diff
@@ -348,11 +374,23 @@ func getManagedEnvs(managedFields []v1.ManagedFieldsEntry, manager string, d *sc
 			return nil, err
 		}
 
-		containers := spec["f:containers"].(map[string]interface{})
-		containerName := fmt.Sprintf(`k:{"name":%q}`, d.Get("container").(string))
-		k := containers[containerName].(map[string]interface{})
-		if e, ok := k["f:env"].(map[string]interface{}); ok {
-			envs = e
+		if isInitContainer {
+			containers := spec["f:initContainers"].(map[string]interface{})
+			containerName := fmt.Sprintf(`k:{"name":%q}`, d.Get("init_container").(string))
+
+			k := containers[containerName].(map[string]interface{})
+			if e, ok := k["f:env"].(map[string]interface{}); ok {
+				envs = e
+			}
+		} else {
+			containers := spec["f:containers"].(map[string]interface{})
+			containerName := fmt.Sprintf(`k:{"name":%q}`, d.Get("container").(string))
+
+			k := containers[containerName].(map[string]interface{})
+			if e, ok := k["f:env"].(map[string]interface{}); ok {
+				envs = e
+			}
+
```

Comment:
These two blocks of code are very similar – can you think of a way to make it simpler?